python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -35,6 +35,7 @@
 import dagster._check as check
 from dagster._annotations import public
 from dagster._core.definitions.events import AssetKey
+from dagster._core.definitions.partition import PartitionsDefinition
```

Comment:
I also encountered some 'most likely due to a circular import' issue when importing these. Do you have suggestions how to solve it properly? It's only used in typehint

Reply:
Take a look at the `if TYPE_CHECKING` section below. It's for exactly this case.